src/service/localization.js

```diff
@@ -90,10 +91,16 @@ export function getLanguageCodesFromString(languageCode) {
 export class LocalizationService {
   /**
    * @param {!Window} win
+   * @param {!./viewer-interface.ViewerInterface} viewer
    */
-  constructor(win) {
+  constructor(win, viewer = undefined) {
```

Comment:
Convention is to avoid changing the constructor since runtime services are constructed generically. 

IOW, use `Services.viewerForDoc`. This would require changing `LocalizationService` to be an AmpDoc service rather than window service (just change the implementation in services.js).